eclair-core/src/main/scala/fr/acinq/eclair/blockchain/bitcoind/BitcoinCoreWallet.scala

```diff
@@ -38,8 +38,8 @@ class BitcoinCoreWallet(rpcClient: BitcoinJsonRPCClient)(implicit system: ActorS
     rpcClient.invoke("fundrawtransaction", hex, BitcoinCoreWallet.Options(lockUnspents)).map(json => {
       val JString(hex) = json \ "hex"
       val JInt(changepos) = json \ "changepos"
-      val JDouble(fee) = json \ "fee"
-      FundTransactionResponse(Transaction.read(hex), changepos.intValue(), (fee * 1e8).toLong)
+      val JDecimal(fee) = json \ "fee"
+      FundTransactionResponse(Transaction.read(hex), changepos.intValue(), Satoshi(fee.bigDecimal.scaleByPowerOfTen(8).longValue()))
```

Comment:
I'm curious, why not use btc2satoshi() like it's in BitcoinCoreFeeProvider?

Reply:
I could have but scaleByPowerOfTen was designed for this, and I will update bitcoin-lib to use it too (Btc uses BigDecimal internally), and then we'll have consistent btc->sat conversions everywhere.